apps/channels/models.py

```diff
@@ -43,11 +43,24 @@ def form(self, team: Team):
 
     def extra_form(self, *args, **kwargs):
         from apps.channels import forms
+        from apps.service_providers.models import MessagingProviderType
+
+        channel = kwargs.pop("channel", None)
 
         match self:
             case self.TELEGRAM:
                 return forms.TelegramChannelForm(*args, **kwargs)
             case self.WHATSAPP:
+                if channel and channel.messaging_provider.type == MessagingProviderType.turnio:
+                    webhook_url = absolute_url(
+                        reverse("channels:new_turn_message", kwargs={"experiment_id": channel.experiment.public_id}),
+                        is_secure=True,
+                    )
+                    initial = kwargs.get("initial", {})
+                    initial.setdefault("verify_token", str(uuid.uuid4()))
+                    initial.setdefault("webook_url", webhook_url)
+                    kwargs["initial"] = initial
```

Comment:
What about moving this to the form class itself? It seams reasonable to me to pass the channel to the form constructor for these forms.

You could then also reuse it in the view instead of having mostly the same code there.